From adac1ec7f90c4ac8a4b66b39a63a4fac45d9048a Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 3 Jan 2022 12:32:24 +0530
Subject: [PATCH 09/14] ath12k: WAR for fw crash while accessing ce 15

ath12k_dp_cc_init and ath12k_dp_reoq_lut_setup caused fw crash
also accessing 15th ce ring cause fw assertion

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 4 ++--
 drivers/net/wireless/ath/ath12k/dp.c   | 6 ++++++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 78432ef..e494490 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -47,11 +47,11 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.internal_sleep_clock = false,
 		.regs = &qcn92xx_regs,
 		.host_ce_config = ath12k_host_ce_config_qcn92xx,
-		.ce_count = 15,
+		.ce_count = 14,
 		.target_ce_config = ath12k_target_ce_config_wlan_qcn92xx,
 		.target_ce_count = 9,
 		.svc_to_ce_map = ath12k_target_service_to_ce_map_wlan_qcn92xx,
-		.svc_to_ce_map_len = 18,
+		.svc_to_ce_map_len = 21,
 		.rxdma1_enable = true,
 		.num_rxmda_per_pdev = 1,
 		.rx_mac_buf_ring = false,
diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 28f03cf..a36d8b4 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1102,6 +1102,8 @@ static void ath12k_dp_cc_cleanup(struct ath12k_base *ab)
 	struct sk_buff *skb;
 	int i;
 
+	return;
+
 	if (!dp->spt_info)
 		return;
 
@@ -1348,6 +1350,8 @@ static int ath12k_dp_cc_init(struct ath12k_base *ab)
 	int i, ret = 0;
 	u32 cmem_base;
 
+	return 0;
+
 	INIT_LIST_HEAD(&dp->rx_desc_free_list);
 	INIT_LIST_HEAD(&dp->rx_desc_used_list);
 	spin_lock_init(&dp->rx_desc_lock);
@@ -1412,6 +1416,8 @@ static int ath12k_dp_reoq_lut_setup(struct ath12k_base *ab)
 {
 	struct ath12k_dp *dp = &ab->dp;
 
+	return 0;
+
 	dp->reoq_lut.vaddr = dma_alloc_coherent(ab->dev,
 						DP_REOQ_LUT_SIZE,
 						&dp->reoq_lut.paddr,
-- 
2.7.4

